server/src/lsp-server.spec.ts

```diff
@@ -99,6 +100,40 @@ describe('completion', () => {
         });
     }).timeout(10000);
 
+    it.only('deprecated by JSDoc', async () => {
+        const doc = {
+            uri: uri('bar.ts'),
+            languageId: 'typescript',
+            version: 1,
+            text: `
+            /**
+             * documentation
+             * @deprecated for a reason
+             */
+            export function foo() {
+                console.log('test')
+            }
+
+            foo(); // call me
+            `
+        };
+        server.didOpenTextDocument({
+            textDocument: doc
+        });
+        const pos = position(doc, 'foo(); // call me');
+        const proposals = await server.completion({
+            textDocument: doc,
+            position: pos
+        }) as TSCompletionItem[];
+        const item = proposals.filter(i => i.label === 'foo')[0];
+        const resolvedItem = await server.completionResolve(item);
+        assert.isTrue(resolvedItem.detail !== undefined, JSON.stringify(resolvedItem, undefined, 2));
```

Comment:
Why would we stringify `undefined` if that assertion fails? Some debug left over code?

Also `assert.isDefined` is a better option here.

Reply:
Ah, I'm not used to chai and was looking for `assert.isDefined`. This whole line is copied from another test, so it's old.